scratchapi2/front.py

```diff
@@ -16,93 +16,115 @@ class FrontPage
+
+    _instance = None
+
+    def __new__(cls, *args, **kwargs):
+        """Enforce this class as a singleton."""
+        if not cls._instance:
+            cls._instance = super().__new__(cls, *args, **kwargs)
+        return cls._instance
+
+    def __init__(self, api_url="https://api.scratch.mit.edu/"):
+        """Initialize the object with the API URL."""
+        self.api_url = api_url
 
     def __repr__(self):
-        """ Always return <ScratchAPI2 FrontPage> """
-        return "<ScratchAPI2 FrontPage>"
+        """Represent the FrontPage."""
+        return "<SFrontPage>"
 
     __str__ = __repr__
 
-    @staticmethod
-    def news(limit=3, offset=0):
-        """ Get Scratch News """
-        result = requests.get("https://api.scratch.mit.edu/news?limit={0}&offset={1}".format(limit, offset)).json()
-        return_values = []
+    def _request(self, path, *opts, api_url=None):
+        """Internal method to request data from the API."""
+        return requests.get(
+            (api_url or self.api_url)
+            + path.format(*opts)
+        ).json()
+
+    def news(self, limit=3, offset=0):
+        """Get Scratch news."""
+        result = self._request('news?limit={0}&offset={1}', limit, offset)
+        News = type('News', (GenericData, object), {
+            '_repr_str': '<News {newsid}>',
+            'newsid': None,
+        })
         for item in result:
-            return_values.append(GenericData(
-                repr_value="<GenericData::News {0}>".format(item["id"]),
+            yield News(
                 newsid=item["id"],
                 timestamp=item["stamp"],
                 title=item["headline"],
                 url=item["url"],
                 image=item["image"],
                 description=item["copy"]
-                )
-                                )
-        return return_values
-
-    @staticmethod
-    def featured_projects():
-        """ Get Featured Projects """
-        result = requests.get("https://api.scratch.mit.edu/proxy/featured").json()["community_featured_projects"]
-        projects = []
+            )
+
+    def featured_projects(self):
+        """Get featured Projects."""
+        result = self._request('proxy/featured')["community_featured_projects"]
         for item in result:
-            projects.append(Project(item["id"], getinfo=False))
-        return projects
-
-    @staticmethod
-    def new_projects():
-        """ Get New Projects """
-        warn("The recent projects row was removed. This API won't continue supporting in the future.", PendingDeprecationWarning)
-        result = requests.get("https://api.scratch.mit.edu/proxy/featured").json()["community_newest_projects"]
-        projects = []
+            yield Project(item["id"], getinfo=GETINFO)
+
+    def new_projects(self):
+        """Get new Projects."""
+        warn("The recent projects row was removed. "
+             "This API endpoint may disappear in the future.",

```

Comment:
This takes advantage of string literal concatenation:
```python
("string 1"
 "string 2")
```
is equivalent to
```python
"string 1string 2"
```
but it only works for string literals, so you CANNOT do this:
```python
var1 = "string 1"
var2 = "string 2"
print(var1 var2)
```